server/src/test/java/org/opensearch/index/compositeindex/datacube/startree/builder/AbstractStarTreeBuilderTests.java

```diff
@@ -2143,18 +2835,37 @@ public void testMergeFlowWithEmptyFieldsInOneSegment() throws IOException {
          [4, 4] | [4]
          [null, 5] | [5]
          */
+        builder.build(starTreeDocumentIterator, new AtomicInteger(), docValuesConsumer);
         int count = 0;
-        while (starTreeDocumentIterator.hasNext()) {
+        for (StarTreeDocument starTreeDocument : builder.getStarTreeDocuments()) {
             count++;
```

Comment:
This removes the count assertion altogether. That is essential to know the results post sortAndReduce

Should we split build into two methods where one appends the star tree docs and the other method does the rest ? ( or more if it helps testing )

Reply:
added the assertion back